spacy_llm/tasks/entity_linker/task.py

```diff
@@ -0,0 +1,240 @@
+from typing import Any, Callable, Dict, Iterable, List, Optional, Tuple, Type
+
+import jinja2
+from spacy import Language
+from spacy.pipeline import EntityLinker
+from spacy.tokens import Doc, Span
+from spacy.training import Example
+
+from ...compat import Self
+from ...ty import FewshotExample, Scorer, TaskResponseParser
+from ..builtin_task import BuiltinTask
+from ..templates import read_template
+from .ty import CandidateSelector, Entity, InitializableCandidateSelector
+
+DEFAULT_EL_TEMPLATE_V1 = read_template("entity_linker.v1")
+
+
+class EntityLinkerTask(BuiltinTask):
+    def __init__(
+        self,
+        parse_responses: TaskResponseParser[Self],
+        prompt_example_type: Type[FewshotExample[Self]],
+        prompt_examples: Optional[List[FewshotExample[Self]]],
+        template: str,
+        scorer: Scorer,
+    ):
+        """Default entity linking task.
+
+        parse_responses (TaskResponseParser[Self]): Callable for parsing LLM responses for this task.
+        prompt_example_type (Type[FewshotExample[Self]]): Type to use for fewshot examples.
+        prompt_examples (Optional[List[FewshotExample[Self]]]): Optional list of few-shot examples to include in prompts.
+        template (str): Prompt template passed to the model.
+        scorer (Scorer): Scorer function.
+        """
+        super().__init__(
+            parse_responses=parse_responses,
+            prompt_example_type=prompt_example_type,
+            template=template,
+            prompt_examples=prompt_examples,
+        )
+        self._scorer = scorer
+        self._candidate_selector: Optional[CandidateSelector] = None
+
+        # Exclude mentions without candidates from prompt, if set.
+        self._auto_nil = True
```

Comment:
The comment says "if set", but is there a way to unset it?

Reply:
No, this is only a debugging internal used in the tests to confirm the mechanism behaves as expected. I don't assume that we want users to be able to modify this behavior.